public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -58,9 +65,14 @@ export class AppChromeService {
   }
 
   public update(update: Partial<AppChromeState>) {
+    const returnToPrevious = getReturnToPrevious();
```

Comment:
Hmm. Why does this read this value from session storage on every state update? I don't think this is great as it leads to unexpected state updates of the returnToPrevious values - setting an unrelated property may lead to the returnToPrevious value reading from session storage again and changing.

Reply:
i think the problem is that plugins don't have access to `AppChromeService`... all they can call is `setReturnToPrevious` which will set it in session storage and then `update` gets called as a side effect of changing route

Reply:
`@grafana/runtime` should expose methods from the _runtime_ (see how backendsrv is made available through there) to plugins

Reply:
yeah so you'd have to expose a `setAppChromeService` method, which will set the singleton in runtime. then get/setReturnToPrevious can directly call `AppChromeService.update()`